module-4/backend/week_four.md

```diff
@@ -5,18 +5,31 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+* My favorite project management tool is waffle, reason being is that I am more familiar with it than PT but as my exposure to PT grows that may change.
 * Why do we create staging environments?
+* Staging environments are created to allow features to be added to the final product, to check how it runs live without using the final production site in case that feature has a negative impact on the product.
 * What are the characteristics of a good README (in your opinion)?
+* For me, good charactersitics of a good README are for that README to be broken down into sections, that's the biggest one. The sections depend on the app but generally I would have Description, Installing/Getting Started and Usage sections as the bare minimum. If alterations need to be made to the code then give directions of what file its going to happen in. 
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+* One main improvement that I make to future code is spacing it appropriately where there is not a lot of text bunched in together.
 * What are some basic security concerns to be aware of when building applications?
+* Basic security concerns to be aware of are having your fields available to take any input and not being strong. If a user where to make a post request using my new user field and my params werent secure then they have the availability to designate themselves whatever role they wish.
 * Why is continuous integration helpful/important?
+* It helps in spotting problems that may have been missed by yourself or the team.
 * What are some continuous integration tools?
+* A good integration tool is Travis.
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+* Good git workflow is making small commits, checking out a branch for each issue.
 * What are the four fundamental concepts of object oriented programming?
+* The four funamental concepts of object oriented programming are Encapsulation, Abstraction, Polymorphism and Inheritence. Encapsulation is the process of combining data and functions into a single unit called a class. Here the data is not accessed directly, it is accessed through the functions inside of that class. Abstraction is the process of showing only relevant data and hiding the unnecessary details from the user. Polymorphism is the ability for data to be represented in many different types. We can send the same data to different objects and get different results. Inheritence is a relations between two classes. A child class inherits all the features of its parent class. Those methods inherited can be overridden in the child and new logic can be added.
 * What's a module in Ruby and what's the difference between a class and a module?
+* Modules are a way of grouping together methods, classes, and constants. A module cannot be used to create an object. Instead it must be mixed in with a class, then its contents are availabe to be used in that class.
```

Comment:
a module isn't a way to group together classes. a module interacts with a class by mixing it into the class. a module encapsulates behavior to be used within a class or on its own 👍